homeassistant/components/reolink/binary_sensor.py

```diff
@@ -0,0 +1,172 @@
+"""This component provides support for Reolink binary sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+
+from reolink_aio.api import (
+    FACE_DETECTION_TYPE,
+    PERSON_DETECTION_TYPE,
+    PET_DETECTION_TYPE,
+    VEHICLE_DETECTION_TYPE,
+)
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ReolinkData
+from .const import DOMAIN
+from .entity import ReolinkCoordinatorEntity
+from .host import ReolinkHost
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class ReolinkBinarySensorDescription(BinarySensorEntityDescription):
+    """A class that describes binary sensor entities."""
+
+    icon: str = "mdi:motion-sensor"
+    icon_off: str = "mdi:motion-sensor-off"
+    value: Callable[[ReolinkHost, int | None], bool | None] = lambda host, ch: None
+    supported: Callable[[ReolinkHost, int | None], bool] = lambda host, ch: True
+
+
+BINARY_SENSORS = (
+    ReolinkBinarySensorDescription(
+        key="motion",
+        name="Motion",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda host, ch: host.api.motion_detected(ch),
+    ),
+    ReolinkBinarySensorDescription(
+        key=FACE_DETECTION_TYPE,
+        name=FACE_DETECTION_TYPE,
+        icon="mdi:face-recognition",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda host, ch: host.api.ai_detected(ch, FACE_DETECTION_TYPE),
+        supported=lambda host, ch: host.api.ai_supported(ch, FACE_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorDescription(
+        key=PERSON_DETECTION_TYPE,
+        name=PERSON_DETECTION_TYPE,
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda host, ch: host.api.ai_detected(ch, PERSON_DETECTION_TYPE),
+        supported=lambda host, ch: host.api.ai_supported(ch, PERSON_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorDescription(
+        key=VEHICLE_DETECTION_TYPE,
+        name=VEHICLE_DETECTION_TYPE,
```

Comment:
Shouldn't this be capitalized instead of the lower case constant value?

```suggestion
        name="Vehicle",
```